### ISSUE TITLE
[bug] Writing to hdf5 file fails with run_multinest and run_ultranest.

### ISSUE THREAD

**Issue by nenasedk** [tomasstolker/species#63]
species == 0.5.5
h5py==3.8.0
python~=3.10.10 and similar

When running a grid fit as in the tutorial, the `fit.run_ultranest` and `fit.run_multinest` functions fail to output correctly to the hdf5 file, returning the error `TypeError: Object dtype dtype('O') has no native HDF5 equivalent`.  This error persists across the use of multiple model grids and companion choices.

The printed outputs are reasonable, providing median and max likelihood parameters, and some samples are clearly being written as the `database.get_mcmc_spectra` and `species.plot_posterior` work fine. However, I think this initial output issue might cause problems with the `species.plot_spectrum` function, which fails to fully complete the plot, and instead gives the error `ValueError: zero-size array to reduction operation minimum which has no identity`.

**Comment by tomasstolker** [comment]
Thanks for opening this issue, @nenasedk!

Add which line in (presumably) `fit_model.py` does the error occur?

**Comment by nenasedk** [comment]
I'd need to rerun the retrieval to get the full traceback, but as far as I can tell it's occuring here:

```File [~/python-packages/species/species/analysis/fit_model.py:2102](https://file+.vscode-resource.vscode-cdn.net/Users/nasedkin/Documents/Paper2_HR8799_Notebooks/~/python-packages/species/species/analysis/fit_model.py:2102), in FitModel.run_ultranest(self, tag, min_num_live_points, output, prior)
   2097 if mpi_rank == 0:
   2098     # Writing the samples to the database is only
   2099     # possible when using a single process
   2100     species_db = database.Database()
-> 2102     species_db.add_samples(
   2103         sampler="ultranest",
   2104         samples=samples,
   2105         ln_prob=ln_prob,```

**Comment by tomasstolker** [comment]
Would you mind running once more and adding this line:
`print(key, value, type(key), type(value))`
within the `for` loop at line 1869 of `database.py` (here I think: _~/python-packages/species/species/data/database.py_)

**Comment by tomasstolker** [comment]
Ah you are using the latest version from Github! Excellent!

The bug is related to the new `ext_filter` parameter. An attribute in the HDF5 database can not be set to `None`.

I think it should be fixed now with commit d54160ebfb90ca800fc9ebb982894dcc11f8a101.

If you `pip install` from Github again then it should hopefully work!

**Comment by nenasedk** [comment]
Great, it works now, thanks a lot!
Turns out the downstream bug in `plot_spectrum` has to do with the choice of filters I was using, so it's unrelated to this issue.
